### PR TITLE
Add Dokka documentation to artifacts

### PR DESCRIPTION
**Pull request by tripolkaandrey**
This PR adds Dokka-generated documentation as a publish artifact. 

It is done by using functionality introduced in [this PR in config](https://github.com/SpineEventEngine/config/pull/352).

Also, while updating the config, the outdated `groovy` directory was removed.

## Important details
In the `2.11.0` version of `error_prone_core`, they broke their contract, which led to a compilation error. The `2.10.0` version was forced for all [other](https://mvnrepository.com/artifact/com.google.errorprone/error_prone_core/2.10.0) Error Prone parts to resolve the issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -110,6 +115,36 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>gfm-template-processing-plugin</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>jekyll-template-processing-plugin</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:143]: @sonatype-lift ignore

**Comment by tripolkaandrey** [comment]
Vulnerabilities regarding Dokka dependencies can be fixed only by adding dependency constraints in the [config](https://github.com/SpineEventEngine/config) repository because they originated from transitive dependencies of the precompiled script [`dokka-for-java`](https://github.com/SpineEventEngine/config/blob/master/buildSrc/src/main/kotlin/dokka-for-java.gradle.kts).

The vulnerability of `error_prone_core` is not relevant as this dependency is forced to [`2.10.0`](https://mvnrepository.com/artifact/com.google.errorprone/error_prone_core/2.10.0) which uses [`30.1-jre`](https://mvnrepository.com/artifact/com.google.guava/guava/30.1-jre) with no mentioned vulnerabilities.

File: pom.xml [hunk 2]
```diff
@@ -110,6 +115,36 @@ all modules and does not describe the project structure per-subproject.
     <artifactId>org.jacoco.ant</artifactId>
     <version>0.8.7</version>
   </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:118]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:143]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -110,6 +115,36 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>all-modules-page-plugin</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:123]: @sonatype-lift ignore

File: pom.xml [hunk 4]
```diff
@@ -110,6 +115,36 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-core</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:133]: @sonatype-lift ignore

File: pom.xml [hunk 5]
```diff
@@ -62,7 +62,7 @@ all modules and does not describe the project structure per-subproject.
   <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:62]: @sonatype-lift ignore

File: pom.xml [hunk 6]
```diff
@@ -110,6 +115,36 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-core</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>gfm-template-processing-plugin</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:138]: @sonatype-lift ignore

File: pom.xml [hunk 7]
```diff
@@ -110,6 +115,36 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>all-modules-page-plugin</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.10</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:128]: @sonatype-lift ignore

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:118]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:143]: @sonatype-lift ignore

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:123]: @sonatype-lift ignore

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:133]: @sonatype-lift ignore

File: pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:138]: @sonatype-lift ignore

File: pom.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:128]: @sonatype-lift ignore

File: pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:62]: @sonatype-lift ignore

File: pom.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:128]: @sonatype-lift ignore

File: pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:138]: @sonatype-lift ignore

File: pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:62]: @sonatype-lift ignore

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:133]: @sonatype-lift ignore

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:123]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:143]: @sonatype-lift ignore

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:118]: @sonatype-lift ignore

File: .idea/misc.xml [hunk 8]
```diff
@@ -21,22 +21,49 @@
     <option name="myDefaultNotNull" value="org.checkerframework.checker.nullness.qual.NonNull" />
     <option name="myNullables">
       <value>
-        <list size="3">
-          <item index="1" class="java.lang.String" itemvalue="org.checkerframework.checker.nullness.qual.Nullable" />
-          <item index="2" class="java.lang.String" itemvalue="org.checkerframework.checker.nullness.compatqual.NullableDecl" />
-          <item index="3" class="java.lang.String" itemvalue="org.checkerframework.checker.nullness.compatqual.NullableType" />
+        <list size="14">
```
Comment (armiol) [inline .idea/misc.xml]: Please make sure this file looks more like the [original one](https://github.com/SpineEventEngine/config/blob/master/.idea/misc.xml).

I don't think we really need all these `@Nullable`-related annotations.

**Review by armiol — CHANGES_REQUESTED** [review]
@tripolkaandrey please see my comment. LGTM otherwise.

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:118]: @sonatype-lift ignore

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:123]: @sonatype-lift ignore

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:133]: @sonatype-lift ignore

File: pom.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:62]: @sonatype-lift ignore

File: pom.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:128]: @sonatype-lift ignore

File: pom.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:138]: @sonatype-lift ignore

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tripolkaandrey) [reply pom.xml:143]: @sonatype-lift ignore

**Comment by tripolkaandrey** [comment]
It was closed in favour of a new PR.

**Pull request closed without merging**